src/pip/_internal/commands/debug.py

```diff
@@ -43,6 +47,106 @@ def show_sys_implementation():
         show_value('name', implementation_name)
 
 
+def find_vendor_txt_path():
+    # type: () -> str
+    vendor_txt_path = os.path.join(
+        os.path.dirname(pip_location),
+        '_vendor',
+        'vendor.txt'
+    )
+    assert os.path.exists(vendor_txt_path), \
+        'Could not find vendor.txt file!'
+
+    return vendor_txt_path
+
+
+def create_vendor_txt_map():
+    # type: () -> Dict[str, str]
+    vendor_txt_path = find_vendor_txt_path()
+
+    with open(vendor_txt_path) as f:
+        # Purge non version specifying lines.
+        # Also, remove any space prefix or suffixes (including comments).
+        lines = [line.strip().split(' ', 1)[0]
+                 for line in f.readlines() if '==' in line]
+
+    # Transform into "module" -> version dict.
+    return dict(line.split('==', 1) for line in lines)  # type: ignore
+
+
+def get_module_from_module_name(module_name):
+    # type: (str) -> ModuleType
+
+    # Module name can be uppercase in vendor.txt for some reason...
+    module_name = module_name.lower()
+    # PATCH: setuptools is actually only pkg_resources.
+    if module_name == 'setuptools':
+        module_name = 'pkg_resources'
+
+    __import__(
+        'pip._vendor.{}'.format(module_name),
+        globals(),
+        locals(),
+        level=0
+    )
+    return getattr(pip._vendor, module_name)
+
+
+def get_vendor_version_from_module(module_name):
+    # type: (str) -> str
+
+    module = get_module_from_module_name(module_name)
+    version = getattr(module, '__version__', None)
+
+    # Some vendored modules (like requests) have a __version__.py file.
```

Comment:
Hmm, this should not be needed thanks to https://github.com/pypa/pip/blob/327a31536ff71b99e5aa454bd325b3daf75b666d/src/pip/_vendor/requests/__init__.py#L110

And indeed:
```
>>> import requests
>>> requests.__version__
'2.23.0'
>>> from pip._vendor.requests import __version__
>>> __version__
'2.22.0'
```

Reply:
I get the same output as you from a python shell, but when running `python  src/pip debug` on a debundled pip I get the following output (when I comment out these two lines):

`requests==<module 'pip._vendor.requests.__version__' from '/usr/lib/python3/dist-packages/requests/__version__.py'> (CONFLICT: vendor.txt suggests version should be 2.22.0)`

Reply:
There is something strange.

With my debundled pip, this is what I get when mimicking your code:
```
$ python -s
Python 3.8.2 (default, Feb 26 2020, 22:21:03) 
[GCC 9.2.1 20200130] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip._vendor
>>> __import__('pip._vendor.requests', globals(), locals(), level=0)
<module 'pip' from '/usr/lib/python3.8/site-packages/pip/__init__.py'>
>>> mod = getattr(pip._vendor, 'requests')
>>> getattr(mod, '__version__', None)
'2.23.0'
>>> mod
<module 'requests' from '/usr/lib/python3.8/site-packages/requests/__init__.py'>
```
What is different in your debundled pip ?

Reply:
What python version are you using ?

Reply:
I use ubuntu 18.04 LTS

```bash
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip._vendor
>>> __import__('pip._vendor.requests', globals(), locals(), level=0)
<module 'pip' from '/usr/lib/python3/dist-packages/pip/__init__.py'>
>>> mod = getattr(pip._vendor, 'requests')
>>> getattr(mod, '__version__', None)
<module 'pip._vendor.requests.__version__' from '/usr/share/python-wheels/requests-2.18.4-py2.py3-none-any.whl/requests/__version__.py'>
>>> mod
<module 'requests' from '/usr/share/python-wheels/requests-2.18.4-py2.py3-none-any.whl/requests/__init__.py'>
>>> ```

Reply:
Debian patches everything. :(

Reply:
I confirm the behavior in 18.04:
```
$ docker run -ti --rm ubuntu-pip-18.04 bash
root@1f2322cbedab:/# python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip
>>> import pip._vendor
>>> __import__('pip._vendor.requests', globals(), locals(), level=0)
<module 'pip' from '/usr/lib/python3/dist-packages/pip/__init__.py'>
>>> mod = getattr(pip._vendor, 'requests')
>>> mod
<module 'requests' from '/usr/share/python-wheels/requests-2.18.4-py2.py3-none-any.whl/requests/__init__.py'>
>>> mod.__version__
<module 'pip._vendor.requests.__version__' from '/usr/share/python-wheels/requests-2.18.4-py2.py3-none-any.whl/requests/__version__.py'>
>>> pip.__version__
'9.0.1'
```
but apparently they seem to have fixed their patching in subsequent version so I think we should be able to drop this special case.
```
root@86a740f3dd26:/# python3
Python 3.8.2 (default, Mar 13 2020, 10:14:16) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pip._vendor
>>> __import__('pip._vendor.requests', globals(), locals(), level=0)
<module 'pip' from '/usr/lib/python3/dist-packages/pip/__init__.py'>
>>> mod = getattr(pip._vendor, 'requests')
>>> mod.__version__
'2.22.0'
>>> mod
<module 'requests' from '/usr/share/python-wheels/requests-2.22.0-py2.py3-none-any.whl/requests/__init__.py'>
>>> pip.__version__
'20.0.2'
```

Reply:
Aren't we supposed to keep supporting active versions?
Ubuntu LTS 18.04 with unupdated python 3 is pretty common, at least in my opinion

Reply:
We keep supporting Ubuntu 18.04.
And indeed running `python3 -m pip  install https://github.com/NoahGorny/pip/archive/pip-debug-print-vendored-versions.zip`, and commenting the `if isinstance(version, ModuleType):` in `/usr/local/lib/python3.6/dist-packages/pip/_internal/commands/debug.py` we still the expected:
```
vendored library versions:
appdirs==1.4.3
CacheControl==0.12.6
colorama==0.4.3
contextlib2==0.6.0 (Unable to locate actual module version, using vendor.txt specified version)
distlib==0.3.0
distro==1.4.0 (Unable to locate actual module version, using vendor.txt specified version)
html5lib==1.0.1
ipaddress==1.0.23
msgpack==0.6.2 (Unable to locate actual module version, using vendor.txt specified version)
packaging==20.1
pep517==0.7.0
progress==1.5
pyparsing==2.4.6
pytoml==0.1.21 (Unable to locate actual module version, using vendor.txt specified version)
requests==2.22.0
certifi==2019.11.28
chardet==3.0.4
idna==2.8
urllib3==1.25.7
retrying==1.3.3 (Unable to locate actual module version, using vendor.txt specified version)
setuptools==44.0.0 (Unable to locate actual module version, using vendor.txt specified version)
six==1.14.0
webencodings==0.5.1 (Unable to locate actual module version, using vendor.txt specified version)
```
output.

Ubuntu 18.04 is using pip 9 and won't upgrade it to the upcoming 20.1 version: its debundled pip will never be affected by this change and more modern Ubuntu versions like 20.04 (and certainly the upcoming 20.10) does not seem affected hence my recommendation to drop the `if isinstance(version, ModuleType):`.

Reply:
I see, removed the part. Thanks a lot for the feedback @xavfernandez, you have been very helpful!